### PR TITLE
Changing string column format: strings should only be read if the column is not empty. Also: Bug fixes

### PR DESCRIPTION
**Pull request by tbrown1979**

### CODE REVIEW

**Comment by tbrown1979** [comment]
retest this please

File: src/main/scala/DefaultCSVFormats.scala [hunk 1]
```diff
@@ -11,7 +11,9 @@ trait CSVColumnFormat[T] {
 
 object CSVColumnFormat {
   implicit object StringColumnFormat extends CSVColumnFormat[String] {
-    def read(c: String, lineNumber: Int) = c.successNel[CSVConversionError]
+    def read(c: String, lineNumber: Int) =
+      if (c.isEmpty) failure[String](c, lineNumber, "empty column")
+      else c.successNel[CSVConversionError]
   }
```
Comment (adamdecaf) [inline src/main/scala/DefaultCSVFormats.scala]: We should call `c.trim.isEmpty` I think. Or `c.blank` (if it exists, I can't remember).

Comment (tbrown1979) [inline src/main/scala/DefaultCSVFormats.scala]: Good call!

**Comment by zmccoy** [comment]
:+1:

**Comment by adamdecaf** [comment]
Pushed up a change to test for non-empty but whitespace-only strings.

**Comment by tbrown1979** [comment]
Thanks for that @adamdecaf. I should have done that when I made the change.

**Comment by tbrown1979** [comment]
Since this wasn't merged yet I just decided to add two fixes to it. Here's the context:

A file like this would give a NullPointerException (read: notice the trailing comma in the header):
```
Zipcode, State, City,
50613, Iowa, Cedar Falls
```
A `null` was being inserted after that trailing comma (by the CSV library we use). I changed it so that when a column is being validated it converts all `null`s to `""`. This was already being done for all rows except the header so it wasn't a big change. 

The other problem that is fixed was matching on headers with `.contains`. This caused some issues with a csv like so: 
```
State, State Abbreviation
Iowa, IA
```
`State` would match when searching for `State Abbreviation`. This would just bug out and always return a Failure because it couldn't correctly identify headers. Now it matches exactly with `==`. To me this seems like the best decision. 

I added two new tests to check for these issues in the future.

**Pull request merged**